### DISCUSSION TITLE
How to add a new filter

### DISCUSSION THREAD

**Discussion by Couleyr** [mboquien/cigale#57] [Q&A]
Dear prof., I attempt to add the new filters (PanSTARRS, grizy bands) to the database, but when I rebuilt it, I got the following error:
pcigale.data.DatabaseInsertError: The filter is already in the base
![截屏2023-06-26 18 02 06](https://github.com/mboquien/cigale/assets/103244840/bd80177b-2a29-4338-8a21-7c3f620de2e3)

my pakgs is named cigale-xray, and the filter file I added is as follows.
![截屏2023-06-26 18 10 43](https://github.com/mboquien/cigale/assets/103244840/b9d1e80a-cd6a-4c84-8669-312024ceee2b)

**Comment by Couleyr** [comment]
However, when I run the command "pcigale-filters add filter_file", it seems that I have successfully added it. And I can find this filter in filters list.
![截屏2023-06-26 18 59 20](https://github.com/mboquien/cigale/assets/103244840/5e8e0658-b3e7-436b-8d11-8dfca6b46eb5)
The filter file I added is located in the main directory (where cigale-xray is located) and not in ./database_builder/filters/, I don't know if there will be any problems with it.

**Reply by Guang91** [reply]
Yes, you can always add new filters with "pcigale-filters add". It doesn't matter where your filter file is.

**Comment by Couleyr** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Okay, thank you very much for replying.
            
        
    Best,Peibin Chen
    
        2023/7/1
    
      
    
        
                    
            
                
                    
                            
                                
                            
                            
                                34820210156868
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
 
    

  
    ---- Replied Message ----
  
  
    
      
        
         From 
        
        
            ***@***.***>
            
        
      
      
        
         Date 
        
        
        07/1/2023 18:09
        
      
      
        
         To 
        
        
         
          
            ***@***.***>
            
          
        
      
      
        
         Cc 
        
        
          
            Peibin ***@***.***>
            ,
          
          
            ***@***.***>
            
          
        
      
      
        
         Subject 
        
        
              Re: [mboquien/cigale] How to add a new filter (Discussion #57)
        
      
    
  
  
Yes, you can always add new filters with "pcigale-filters add". It doesn't matter where your filter file is.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
